Cabal-syntax/src/Distribution/Fields/Lexer.hs

```diff
@@ -454,35 +498,56 @@ alexScanUser user__ input__ (I# (sc))
 alex_scan_tkn user__ orig_input len input__ s last_acc =
   input__ `seq` -- strict in the input
   let
-  new_acc = (check_accs (alex_accept `quickIndex` (I# (s))))
+  new_acc = (check_accs (alex_accept `quickIndex` IBOX(s)))
   in
   new_acc `seq`
   case alexGetByte input__ of
      Nothing -> (new_acc, input__)
      Just (c, new_input) ->
-
-      case fromIntegral c of { (I# (ord_c)) ->
+#ifdef ALEX_DEBUG
+      trace ("State: " ++ show IBOX(s) ++ ", char: " ++ show c) $
+#endif
+      case fromIntegral c of { IBOX(ord_c) ->
         let
                 base   = alexIndexInt32OffAddr alex_base s
-                offset = (base +# ord_c)
-                check  = alexIndexInt16OffAddr alex_check offset
+                offset = PLUS(base,ord_c)
 
-                new_s = if GTE(offset,0#) && EQ(check,ord_c)
+                new_s = if GTE(offset,ILIT(0))
+                          && let check  = alexIndexInt16OffAddr alex_check offset
+                             in  EQ(check,ord_c)
```

Comment:
This comes from this Alex commit: [haskell/alex@`e96c347` (#224)](https://github.com/haskell/alex/pull/224/commits/e96c3478f64bd34b51d313911341c54e90898faf)

I do not see how this should have introduced the problems we are seeing.
`GTE` `ILIT` and `EQ` are macros, yes, but they do not have side effects, so moving the `check` binding into a local `let` should not be a semantic change except making things lazier.